src/WebApiContrib.Core.Versioning/Strategies/IVersionStrategy.cs

```diff
@@ -15,6 +15,6 @@ public interface IVersionStrategy
         /// <param name="context">The <see cref="HttpContext"/> of the current request.</param>
         /// <param name="routeData">The <see cref="RouteData"/>of the current request.</param>
         /// <returns>A version integer, or <c>null</c> if a version cannot be determined.</returns>
-        int? GetVersion(HttpContext context, RouteData routeData);
+        VersionContext GetVersion(HttpContext context, RouteData routeData);
```

Comment:
Can we make `VersionContext` a `struct` and make this `VersionContext?`. Maybe we should call it `VersionResult`?